### PR TITLE
api: Allow custom OSes to be used in VOD

### PR DESCRIPTION
**Pull request by victorges**
**What does this pull request do? Explain your changes. (required)**
This is to make a few changes in the API to allow Storj to be used for VOD
with limited capabilities (direct and resumable uploads are currently broken).

**Specific updates (required)**
Allow using custom OS on assets and recordings.

## -

- **How did you test each of these updates (required)**
Uploaded assets to Storj!!
https://www.loom.com/share/9dcdebc0f2dc463a9beab1ac03a8b4d1

**Does this pull request close any open issues?**
Implements #1267 

**Screenshots (optional):**
https://www.loom.com/share/9dcdebc0f2dc463a9beab1ac03a8b4d1

**Checklist:**

- [x] I have read the **CONTRIBUTING** document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/asset.ts [hunk 1]
```diff
@@ -288,15 +310,15 @@ app.use(
     };
 
     if (Array.isArray(data)) {
-      return data.map(toExternalAsset);
+      return Promise.all(data.map(toExternalAsset));
     }
     if ("id" in data) {
       return toExternalAsset(data);
```
Comment (gioelecerati) [inline packages/api/src/controllers/asset.ts:316]: Shouldn't also this be `await toExternalAsset(data)` ?

**Review by gioelecerati — APPROVED** [review]
LGTM

File: packages/api/src/controllers/asset.ts (hunk 1 above)
Comment (victorges) [reply packages/api/src/controllers/asset.ts:316]: On JS you can wrap promises like `Promise<Promise<Promise<string>>>` and they will still work the same. 
i.e. you can also return a promise from an async function, and that will still work for the caller. 

note1 : TypeScript would complain if this wasn't valid, so we have an extra guarantee there 😌 
note2: Maybe the "promise unwrap" only happens for `async` functions. Maybe if you actually have a variable that is a `Promise<Promise<string>>` you'd have to manually unwrap it into a `Promise<string>` somehow. IDK. Maybe it's just magic tho lol

**Comment by victorges** [comment]
Closing this as we don't really want the `playbackRecordingObjectStoreId`. We will be migrating to catalyst where everything is under the same OS. We might need a part of this later (using OS public URL for playback for external OSes), but it's too small of a change to include this whole PR, we can refer back to this when we want to implement that.

**Pull request closed without merging**